packages/@glimmer/interfaces/lib/managers/component.d.ts

```diff
@@ -0,0 +1,48 @@
+import { Capabilities } from './capabilities';
+import { Arguments } from '../runtime';
+
+export interface ComponentCapabilitiesVersions {
+  '3.4': {
+    asyncLifecycleCallbacks?: boolean;
+    destructor?: boolean;
+  };
+
+  '3.13': {
+    asyncLifecycleCallbacks?: boolean;
+    destructor?: boolean;
+    updateHook?: boolean;
+  };
```

Comment:
I definitely have no objection here (and this isn't a blocking concern), but we **need** to figure out how we are going to pick these numbers in the future.

Reply:
Agreed, wholeheartedly. I think honestly just versioning each API independently is probably a decent solution, as long as we document them in the API docs. It means users will have to do a bit of extra looking-up to figure out what capabilities are available (and supported), but that's a given with this APIs anyways.